mailet/crypto/src/main/java/org/apache/james/transport/matcher/IsX509CertificateSubject.java

```diff
@@ -69,12 +72,12 @@ public void init() throws MessagingException {
     public Collection<MailAddress> match(Mail mail) throws MessagingException {
         List<X509Certificate> certificates;
         
-        Object obj = mail.getAttribute(sourceAttribute);
-        if (obj != null) {
-            if (obj instanceof X509Certificate) {
-                certificates = Collections.singletonList((X509Certificate)obj);
+        Optional<?> obj = AttributeUtils.getAttributeValueFromMail(mail, AttributeName.of(sourceAttribute));
+        if (obj.isPresent()) {
+            if (obj.get() instanceof X509Certificate) {
```

Comment:
Maybe we can extract that piece of code in order to use `map` and use Optionals in a functional way